### DISCUSSION TITLE
Placing a condition on ModelCheckpoint Callback

### DISCUSSION THREAD

**Discussion by skrish13** [Lightning-AI/pytorch-lightning#13337] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule] [answered]
For example: How to save checkpoints ONLY when valset accuracy is greater than some specific threshold?

If this is not possible using parameters in ModelCheckpoint callback, how can we extend ModelCheckpoint to achieve this?

**Comment by rohitgr7** [comment, marked as answer]
you can do:
```py
class CustomModelCheckpoint(ModelCheckpoint):
    def on_validation_end(self, trainer, pl_module):
        score = trainer.callback_metrics[self.monitor]
        if score > some_threshold:
            super().on_validation_end(trainer, pl_module)


trainer = Trainer(callbacks=[CustomModelCheckpoint(...)])
```
